src/main/java/org/mitre/synthea/export/BB2RIFExporter.java

```diff
@@ -3254,89 +3246,108 @@ private long exportSNF(Person person, long startTime, long stopTime) throws IOEx
             default:
               // Override mapped REV_CNTR when a PPS code is present and not a medication
               // SNF claim paid under PPS submitted as type of bill (TOB) 21X
               fieldValues.put(SNF.REV_CNTR, "0022");
-              fieldValues.put(SNF.REV_CNTR_UNIT_CNT, Integer.toString(lineItem.getCount()));
+              fieldValues.put(SNF.REV_CNTR_UNIT_CNT, Integer.toString(revCntrCount));
               fieldValues.remove(SNF.REV_CNTR_NDC_QTY);
               fieldValues.remove(SNF.REV_CNTR_NDC_QTY_QLFR_CD);
               break;
           }
 
           fieldValues.put(SNF.CLM_LINE_NUM, Integer.toString(claimLine++));
-          fieldValues.put(SNF.REV_CNTR_RATE_AMT,
-              String.format("%.2f", lineItem.cost
-                      .divide(BigDecimal.valueOf(Integer.max(1, days)), RoundingMode.HALF_EVEN)
-                      .setScale(2, RoundingMode.HALF_EVEN)));
-          fieldValues.put(SNF.REV_CNTR_TOT_CHRG_AMT,
-              String.format("%.2f", lineItem.cost));
-          fieldValues.put(SNF.REV_CNTR_NCVRD_CHRG_AMT,
-              String.format("%.2f", lineItem.copayPaidByPatient
-              .add(lineItem.deductiblePaidByPatient).add(lineItem.patientOutOfPocket)));
-          if (lineItem.patientOutOfPocket.compareTo(Claim.ZERO_CENTS) == 0
-                  && lineItem.deductiblePaidByPatient.compareTo(Claim.ZERO_CENTS) == 0) {
-            // Not subject to deductible or coinsurance
-            fieldValues.put(SNF.REV_CNTR_DDCTBL_COINSRNC_CD, "3");
-          } else if (lineItem.patientOutOfPocket.compareTo(Claim.ZERO_CENTS) > 0
-                  && lineItem.deductiblePaidByPatient.compareTo(Claim.ZERO_CENTS) > 0) {
-            // Subject to deductible and coinsurance
-            fieldValues.put(SNF.REV_CNTR_DDCTBL_COINSRNC_CD, "0");
-          } else if (lineItem.patientOutOfPocket.compareTo(Claim.ZERO_CENTS) == 0) {
-            // Not subject to deductible
-            fieldValues.put(SNF.REV_CNTR_DDCTBL_COINSRNC_CD, "1");
-          } else {
-            // Not subject to coinsurance
-            fieldValues.put(SNF.REV_CNTR_DDCTBL_COINSRNC_CD, "2");
-          }
+          setSNFClaimLineCosts(fieldValues, lineItem, Integer.max(1, revCntrCount));
           rifWriters.writeValues(SNF.class, fieldValues);
         }
 
-        if (claimLine == 1) {
-          // If claimLine still equals 1, then no line items were successfully added.
-          // Add a single top-level entry for the total charge
-          fieldValues.put(SNF.CLM_LINE_NUM, Integer.toString(claimLine));
-          fieldValues.remove(SNF.HCPCS_CD);
-          fieldValues.put(SNF.REV_CNTR, "0001"); // Total charge
-          fieldValues.put(SNF.REV_CNTR_UNIT_CNT, "0");
-          rifWriters.writeValues(SNF.class, fieldValues);
-        }
+        // Add a total charge etry
+        setSNFClaimLineCosts(fieldValues, consolidatedClaimLines, 1);
+        fieldValues.put(SNF.CLM_LINE_NUM, Integer.toString(claimLine));
```

Comment:
```suggestion
        fieldValues.put(SNF.CLM_LINE_NUM, Integer.toString(claimLine++));
```
I think you want to add to the claimLine just like in HHA.